zulip/zulip/__init__.py

```diff
@@ -44,6 +45,31 @@
 
 API_VERSTRING = "v1/"
 
+# An optional parameter to `move_topic` and `update_message` actions
+EditPropagateMode = Literal["change_one", "change_all", "change_later"]
+
+# Generally a `reaction_type` parameter whenever an emoji is specified:
+# - optional parameters to actions: `add_reaction`, `remove_reaction`
+# - events: "user_status", "reaction", "message", "update_message"
+# - generally inside each reaction in the `reactions` field of returned message objects
+EmojiType = Literal["realm_emoji", "unicode_emoji", "zulip_extra_emoji"]
+
+# Personally modifiable message flags:
+# - updated by `update_message_flags`, `mark_all_as_read`, `mark_stream_as_read`, `mark_topic_as_read` actions
+# - notified to the user via "update_message_flag" events
+ModifiableMessageFlag = Literal["read", "starred", "collapsed"]
+
+# All possible message flags
+# - generally present in `flags` object of returned message objects
+# - notified to the user via "update_message_flag" events
+MessageFlag = Literal[
+    ModifiableMessageFlag,
+    "mentioned",
+    "wildcard_mentioned",
+    "has_alert_word",
+    "historical",
+]
```

Comment:
I'd encourage you to add a link here to where these are described in the API docs:
https://zulip.com/api/update-message-flags#available-flags

That's good for the reader looking at this library and trying to understand what the semantics are -- in particular, what these flags mean -- because it helps them find where we describe that.

It's also good for a future reader seeking to update the library (perhaps yourself) by providing a quick way to compare against the then-current docs and see if anything's changed.

We try to make a similar practice in the API bindings within zulip-mobile. Discussion here:
https://github.com/zulip/zulip-mobile/blob/main/docs/style.md#api-doc-link